### ISSUE TITLE
There are sth wrong when type the value to input of dates if I give [value] to RangePicker.

### ISSUE THREAD

**Issue by lininu** [ant-design/ant-design#13652]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.11.2

### Environment
Chrome 70.0.3538.110

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/8l1mlvwvzl)

### Steps to reproduce
![Image of Steps to reproduce](https://lininu.github.io/issue-box/src/images/antd_rangepicker-value-issue.gif)

### What is expected?
Noting wrong just like currect example.

### What is actually happening?
After type the dates, it seems that value disappeared.

**Comment by ztplz** [comment]
![image](https://user-images.githubusercontent.com/17755202/50086854-58ef9b00-0239-11e9-8eb6-36bd10df4b49.png)
Comment this line.
![untitled7](https://user-images.githubusercontent.com/17755202/50086875-6d339800-0239-11e9-935a-dba690c93059.gif)

**Comment by lininu** [comment]
@ztplz 
In another case, I need to give it default or dynamic change the value of this picker. So I need to reserve the `value property`

**Comment by ztplz** [comment]
Edit your example and hope to help you.  https://codesandbox.io/s/mooo4p1j0x

**Comment by lininu** [comment]
@ztplz Hi, 
Thank you for your help, but I think it's not the result what I want. I need both of `value property` and the feature of `type dates`.
![image of task](https://lininu.github.io/issue-box/src/images/antd_rangepicker-value-issue-1.gif)
In your example, typing dates are still wrong.

**Comment by ztplz** [comment]
Actually this is a user-friendly design that allows the user to not have to deal with the logic between the input box and the calendar. So if you don't set the value, `Antd` will handle the input and calendar by default. But if you set the value, `Antd` will let you control the display of the time yourself. Here is a [modified example](https://codesandbox.io/s/mooo4p1j0x).

**Comment by lininu** [comment]
> Actually this is a user-friendly design that allows the user to not have to deal with the logic between the input box and the calendar. So if you don't set the value, `Antd` will handle the input and calendar by default. But if you set the value, `Antd` will let you control the display of the time yourself. Here is a [modified example](https://codesandbox.io/s/mooo4p1j0x).

The example you create is not using `immutable`... I thought it's a little be dangerous design pattern. But if you dont think it's an issue, I think I can use your example temporarily.

**Comment by ztplz** [comment]
It is immutable. I did not directly modify the original array when `setState`.

**Comment by lininu** [comment]
> It is immutable. I did not directly modify the original array when `setState`.


I tried to modify your example, first `startIncrease` seems correct.
But second `startDecrease` I modified `startTime` to an new moment object, the value of RangePicker cannot be changed.
So third, `endIncrease` case, I do not use `setState` and call `forceUpdate`. Both of the end value of RangePicker and current state of end value is changed...
![image of sample](https://lininu.github.io/issue-box/src/images/antd_rangepicker-value-issue-2.gif)
[https://codesandbox.io/s/l9j2lpq6qm](url)
